src/gsy_e/models/strategy/energy_parameters/virtual_heat_pump.py

```diff
@@ -0,0 +1,196 @@
+            initial_temp_C: float = ConstSettings.HeatPumpSettings.INIT_TEMP_C,
+            tank_volume_l: float = ConstSettings.HeatPumpSettings.TANK_VOL_L,
+            water_supply_temp_C_profile: Optional[Union[str, float, Dict]] = None,
+            water_supply_temp_C_profile_uuid: Optional[str] = None,
+            water_return_temp_C_profile: Optional[Union[str, float, Dict]] = None,
+            water_return_temp_C_profile_uuid: Optional[str] = None,
+            dh_water_flow_m3_profile: Optional[Union[str, float, Dict]] = None,
+            dh_water_flow_m3_profile_uuid: Optional[str] = None,
+    ):
+        super().__init__(
+            maximum_power_rating_kW, min_temp_C, max_temp_C, initial_temp_C, tank_volume_l)
+
+        self._water_supply_temp_C: [DateTime, float] = EnergyProfile(
+            water_supply_temp_C_profile, water_supply_temp_C_profile_uuid,
+            profile_type=InputProfileTypes.IDENTITY)
+
+        self._water_return_temp_C: [DateTime, float] = EnergyProfile(
+            water_return_temp_C_profile, water_return_temp_C_profile_uuid,
+            profile_type=InputProfileTypes.IDENTITY)
+
+        self._dh_water_flow_m3: [DateTime, float] = EnergyProfile(
+            dh_water_flow_m3_profile, dh_water_flow_m3_profile_uuid,
+            profile_type=InputProfileTypes.IDENTITY)
+
+    def serialize(self):
+        """Return dict with the current energy parameter values."""
+        return {
+            **super().serialize(),
+            "water_supply_temp_C": self._water_supply_temp_C.input_profile,
+            "water_supply_temp_C_uuid": self._water_supply_temp_C.input_profile_uuid,
+            "water_return_temp_C": self._water_return_temp_C.input_profile,
+            "water_return_temp_C_uuid": self._water_return_temp_C.input_profile_uuid,
+            "dh_water_flow_m3_profile": self._dh_water_flow_m3.input_profile,
+            "dh_water_flow_m3_profile_uuid": self._dh_water_flow_m3.input_profile_uuid,
+        }
+
+    def _rotate_profiles(self, current_time_slot: Optional[DateTime] = None):
+        self._water_return_temp_C.read_or_rotate_profiles()
+        self._water_supply_temp_C.read_or_rotate_profiles()
+        self._dh_water_flow_m3.read_or_rotate_profiles()
+        self.state.delete_past_state_values(current_time_slot)
+
+    def _calc_energy_to_buy_maximum(self, time_slot: DateTime) -> float:
+        max_energy_consumption = self._storage_temp_to_energy(self._max_temp_C, time_slot)
+        assert max_energy_consumption > -FLOATING_POINT_TOLERANCE
+        return min(self._max_energy_consumption_kWh, max_energy_consumption)
+
+    def _calc_energy_to_buy_minimum(self, time_slot: DateTime) -> float:
+        max_energy_consumption = self._storage_temp_to_energy(
+            self.state.get_storage_temp_C(time_slot), time_slot)
+        assert max_energy_consumption > -FLOATING_POINT_TOLERANCE
+        return min(self._max_energy_consumption_kWh, max_energy_consumption)
```

Comment:
In order to not get confused later on, how about renaming `max_energy_consumption` into `min_energy_consumption` ?

Reply:
Agreed and renaming